### DISCUSSION TITLE
question about node collection

### DISCUSSION THREAD

**Discussion by Tusenka** [pytest-dev/pytest#13011] [Q&A] [answered]
Why do we need to get __dict__ from parent class? __dict__  is already already loaded from parent object
https://github.com/pytest-dev/pytest/blob/b938e702fb62cbfb51a667ac7f783222509b9134/src/_pytest/python.py#L392
and in which cases there are possible duplicates for seen?

**Comment by RonnyPfannschmidt** [comment, marked as answer]
Some testsuites use subclasses that override some tests

The only save way is to walk the dicts of the complete mro and taking overrides into account
